core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitMutationOperationChecker.java

```diff
@@ -52,51 +55,54 @@ public void check(Mutation mutation) throws ExecutionException {
   }
 
   private void check(Put put) throws ExecutionException {
-    put.getCondition()
-        .ifPresent(
-            condition -> {
-              if (!(condition instanceof PutIf
-                  || condition instanceof PutIfNotExists
-                  || condition instanceof PutIfExists)) {
-                throw new IllegalArgumentException(
-                    "A "
-                        + condition.getClass().getSimpleName()
-                        + " condition is not allowed on Put operation");
-              }
-            });
     TransactionTableMetadata metadata = getTableMetadata(put);
     for (String column : put.getContainedColumnNames()) {
       if (metadata.getTransactionMetaColumnNames().contains(column)) {
         throw new IllegalArgumentException(
             "Mutating transaction metadata columns is not allowed. column: " + column);
       }
     }
-    checkConditionIsNotTargetingMetadataColumns(put, metadata);
+
+    if (!put.getCondition().isPresent()) {
+      return;
+    }
+    MutationCondition condition = put.getCondition().get();
+
+    if (!(condition instanceof PutIf
+        || condition instanceof PutIfNotExists
+        || condition instanceof PutIfExists)) {
+      throw new IllegalArgumentException(
+          "A "
+              + condition.getClass().getSimpleName()
+              + " condition is not allowed on Put operation");
+    }
+    checkConditionIsNotTargetingMetadataColumns(condition, metadata);
+    ConditionChecker conditionChecker = conditionCheckerFactory.create(metadata.getTableMetadata());
+    conditionChecker.check(condition, true);
```

Comment:
I hesitated to rely on the `@VisibleForTesting` pattern, but since the `ConditionCheckerFactory` is only useful for the `ConsensusCommitMutationOperationChecker`, I think you are right. Thank you.
see https://github.com/scalar-labs/scalardb/pull/899/commits/afc2fbec6c4d7c64cf7617d94892a84cfce0e297